airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/attribution_report.py

```diff
@@ -0,0 +1,172 @@
+
+BRAND_REFERRAL_BONUS = "brb_bonus_amount"
+
+METRICS_MAP = {
+    "PERFORMANCE": [
+        "Click-throughs",
+        "attributedDetailPageViewsClicks14d",
+        "attributedAddToCartClicks14d",
+        "attributedPurchases14d",
+        "unitsSold14d",
+        "attributedSales14d",
+        "attributedTotalDetailPageViewsClicks14d",
+        "attributedTotalAddToCartClicks14d",
+        "attributedTotalPurchases14d",
+        "totalUnitsSold14d",
+        "totalAttributedSales14d",
+    ],
+    "PRODUCTS": [
+        "attributedDetailPageViewsClicks14d",
+        "attributedAddToCartClicks14d",
+        "attributedPurchases14d",
+        "unitsSold14d",
+        "attributedSales14d",
+        "brandHaloDetailPageViewsClicks14d",
+        "brandHaloAttributedAddToCartClicks14d",
+        "brandHaloAttributedPurchases14d",
+        "brandHaloUnitsSold14d",
+        "brandHaloAttributedSales14d",
+        "attributedNewToBrandPurchases14d",
+        "attributedNewToBrandUnitsSold14d",
+        "attributedNewToBrandSales14d",
+        "brandHaloNewToBrandPurchases14d",
+        "brandHaloNewToBrandUnitsSold14d",
+        "brandHaloNewToBrandSales14d",
+    ],
+}
+
+
+class AttributionReport(AmazonAdsStream):
+    """
+    This stream corresponds to Amazon Advertising API - Attribution Reports
+    https://advertising.amazon.com/API/docs/en-us/amazon-attribution-prod-3p/#/
+    """
+
+    model = AttributionReportModel
+    primary_key = None
+    data_field = "reports"
+    page_size = 300
+
+    report_type = ""
+    metrics = ""
+    group_by = ""
+
+    _next_page_token_field = "cursorId"
+    _current_profile_id = ""
+
+    REPORT_DATE_FORMAT = "YYYYMMDD"
+    CONFIG_DATE_FORMAT = "YYYY-MM-DD"
+    REPORTING_PERIOD = 90
+
+    def __init__(self, config: Mapping[str, Any], *args, **kwargs):
+        self._start_date = config.get("start_date")
+        self._req_start_date = ""
+        self._req_end_date = ""
+
+        super().__init__(config, *args, **kwargs)
+
+    def _set_dates(self, profile: Profile):
+        new_start_date = pendulum.now(tz=profile.timezone).subtract(days=1).date()
+        new_end_date = pendulum.now(tz=profile.timezone).date()
+
+        if self._start_date:
+            new_start_date = max(self._start_date, new_end_date.subtract(days=self.REPORTING_PERIOD))
+
+        self._req_start_date = new_start_date.format(self.REPORT_DATE_FORMAT)
+        self._req_end_date = new_end_date.format(self.REPORT_DATE_FORMAT)
+
+    @property
+    def http_method(self) -> str:
+        return "POST"
+
+    def read_records(self, *args, **kvargs) -> Iterable[Mapping[str, Any]]:
+        """
+        Iterate through self._profiles list and send read all records for each profile.
+        """
+        for profile in self._profiles:
+            self._set_dates(profile)
+            self._current_profile_id = profile.profileId
+            yield from super().read_records(*args, **kvargs)
```

Comment:
```py
try:
    self._set_dates(profile)
    self._current_profile_id = profile.profileId
    yield from super().read_records(*args, **kvargs)
except Exception:
    # This profile doesn't have any records/access
    self.logger.info(
        "This profile doesn't have any records/access")
```

Reply:
@harshithmullapudi 

Have implemented above changes. Updated the unit tests also. PTAL

Reply:
Thanks a lot.